unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -95,6 +95,34 @@ public static IntPtr gchandle_get_target_v2([NativeWrapperType("uintptr_t")] Int
         return Unsafe.As<object, IntPtr>(ref obj);
     }
 
+    [return: NativeCallbackType("MonoObject*")]
+    public static IntPtr object_isinst([NativeCallbackType("MonoObject*")] IntPtr obj, [NativeCallbackType("MonoClass*")] IntPtr klass)
+    {
+        var instance = Unsafe.As<IntPtr, object>(ref obj);
+        var type = Unsafe.As<IntPtr, Type>(ref klass);
+        Type current = instance.GetType();
+
+        while (current != null)
+        {
+            if (current == type)
+                return obj;
+
+            // TODO : Unclear what the value of this is.  The if above is always hit first preventing rank differences from being taken into consideration
+            // if (current.IsArray && type.IsArray && current.GetArrayRank() == type.GetArrayRank() && current.GetElementType() == type.GetElementType())
+            //     return obj;
+
+            foreach (var iface in current.GetInterfaces())
+            {
+                if (iface == type)
+                    return obj;
+            }
+
+            current = current.BaseType;
+        }
+
+        return nint.Zero;
```

Comment:
The old native implementation returned `NULL` when the object was not of the given type.  Does does anyone see a way to get `mono_object_isinst` to return `NULL` from here?

As this code is now, I have a test failure.  I'm trying to figure out the cleanest way to address this.
```
D:\UnitySrc\dev\unity-runtime\unity\embed_api_tests\main.cpp(513): FAILED:
  CHECK( mono_object_isinst(testobj, klass) != 0 )
with expansion:
  NULL != 0
```

Reply:
I think your error is that you are returning 0/NULL when you should not. That test is asserting for not 0.